### PR TITLE
Add new eks terraform deployment check

### PR DESCRIPTION
**Pull request by benjamb**
Makes use of composable actions to promote code reuse for later workflows per provider/resource.

Includes a filter similar to https://github.com/finos/cloud-service-certification/pull/148 to ensure we only run the workflow whenever relevant files are changed.

Yet to be solved is how to ensure such workflows are run sequentially. Setting `concurrency` is potentially a no-go as it cancels pending pipelines, with no current way to disable that behaviour.

Part of https://github.com/finos/cloud-service-certification/issues/124.

### CODE REVIEW

**Comment by thinkl33t** [comment]
I've modified this to only run the deployment when secrets are available, so a deploy will run when: 

- a PR is opened, or a push is made to a branch 
- AND the pr or push modifies the CI or any part of the EKS terraform
- AND the branch is from inside the FinOS organisation

Validation will run when:

- a PR is opened, or a push is made to a branch 
- AND the pr or push modifies the CI or any part of the EKS terraform

This puts the framework in place for most of the options for future workflows mentioned in #152, allowing the results of the codethink CI sprint to be merged.

File: .github/workflows/aws-eks-verify.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+env:
+  TF_IN_AUTOMATION: true
+  TF_INPUT: false
+  TF_CLI_ARGS: "-no-color"
+  WORKING_DIRECTORY: aws/eks/eks-terraform-scripts
+
+jobs:
+  validate:
+    runs-on: ubuntu-latest
```
Comment (abdullahgarcia) [inline .github/workflows/aws-eks-verify.yml:26]: Can we please use a specific version? 

This is not a best practice.

File: .github/workflows/aws-eks-verify.yml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+          working-directory: ${{ env.WORKING_DIRECTORY }}
+      - name: check for secrets needed for eks deploy
+        id: check-eks-secrets
+        run: |
+          if [ ! -z "${{ secrets.AWS_ACCESS_KEY_ID }}" ]; then
+            echo "::set-output name=eks-ok::true"
+          fi
+
+  ephemeral-deploy:
+    runs-on: ubuntu-latest
```
Comment (abdullahgarcia) [inline .github/workflows/aws-eks-verify.yml:46]: Same here.

**Review by abdullahgarcia — CHANGES_REQUESTED** [review]
Left some comments on best practices.

File: .github/workflows/aws-eks-verify.yml (hunk 1 above)
Comment (thinkl33t) [reply .github/workflows/aws-eks-verify.yml]: Fixed to 20.04, thanks

File: .github/workflows/aws-eks-verify.yml (hunk 2 above)
Comment (thinkl33t) [reply .github/workflows/aws-eks-verify.yml]: Fixed to 20.04, thanks

**Review by abdullahgarcia — APPROVED** [review]
LGTM

**Comment by mcleo-d** [comment]
Well done @thinkl33t and the rest of the Codethink team 🚀

**Pull request merged**